src/Microsoft.Security.Utilities.Core/HighPerformanceScanner.cs

```diff
@@ -0,0 +1,505 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Security.Utilities;
+
+#if NET8_0_OR_GREATER
+using System.Buffers;
+using StringInput = System.ReadOnlySpan<char>;
+#else
+using StringInput = string;
+#endif
+
+/// <summary>
+/// A detection from <see cref="HighPerformanceScanner"/>.
+/// </summary>
+internal readonly record struct HighPerformanceDetection
+{
+    /// <summary>
+    /// The signature that identifies the pattern detected.
+    /// </summary>
+    public string Signature { get; }
+
+    /// <summary>
+    /// The start index in the input where the pattern was detected.
+    /// </summary>
+    public int Start { get; }
+
+    /// <summary>
+    /// The length of the  in the input that matched in the
+    /// </summary>
+    public int Length { get; }
+
+    public HighPerformanceDetection(string signature, int start, int length)
+    {
+        Signature = signature;
+        Start = start;
+        Length = length;
+    }
+}
+
+/// <summary>
+/// A pattern that <see cref="HighPerformanceScanner"/> can scan for. These are
+/// typically the precomputed set derived from the patterns that implement <see
```

Comment:
```suggestion
/// typically use the precomputed set derived from the patterns that implement <see
```

nit…I think i've fixed it? #Resolved

Reply:
I've changed it. That edit doesn't reflect what I wanted to say.